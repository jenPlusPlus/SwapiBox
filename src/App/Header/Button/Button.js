import React from 'react'

const Button = ({ buttonTitle, buttonAction}) => {
  return (
    <div>
      <a href={`/${buttonTitle.toLowerCase()}`}>
        <button>
          {buttonTitle}
        </button>
      </a>
    </div>
  );
};

export default Button;
