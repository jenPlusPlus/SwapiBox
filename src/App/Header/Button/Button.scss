.button-wrapper{
  width: 10%;
  display: inline-block;
  margin: 3%;
}

.active {
  background-color: #e0531e;
  border: none;
  color: #e0d1a8;
  width: 100%;
  }

  .button-inactive{
    background-color: #672807;
    border: none;
    color: #e0d1a8;
    width: 100%;
  }
