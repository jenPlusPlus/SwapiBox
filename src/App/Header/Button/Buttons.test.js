import React from 'react';
import ReactDOM from 'react-dom';
import Button from './Button';
import fetchMock from 'fetch-mock';
import renderer from 'react-test-renderer';
import { withRouter } from 'react-router';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router, Route } from 'react-router-dom';

describe('CARD', () => {
  const mkFun = jest.fn();
  let wrapper = shallow(<Button
                          isActive={false}
                          buttonTitle={'Favorites'}
                          numFavorites={0}
                          apiCall={mkFun}
                              />);

it('should match the Card snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});
