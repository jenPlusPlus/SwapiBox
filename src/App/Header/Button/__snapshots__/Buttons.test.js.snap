// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD should match the Card snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    apiCall={[Function]}
    buttonTitle="Favorites"
    isActive={false}
    numFavorites={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="/favorites"
>
        <button
                className="button button-inactive Favorites-button"
        >
                Favorites
                <span>
                         : 
                </span>
                0
        </button>
</Link>,
      "className": "button-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <button
          className="button button-inactive Favorites-button"
>
          Favorites
          <span>
                     : 
          </span>
          0
</button>,
        "replace": false,
        "to": "/favorites",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites",
            <span>
               : 
</span>,
            0,
          ],
          "className": "button button-inactive Favorites-button",
        },
        "ref": null,
        "rendered": Array [
          "Favorites",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": " : ",
            },
            "ref": null,
            "rendered": " : ",
            "type": "span",
          },
          0,
        ],
        "type": "button",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="/favorites"
>
          <button
                    className="button button-inactive Favorites-button"
          >
                    Favorites
                    <span>
                               : 
                    </span>
                    0
          </button>
</Link>,
        "className": "button-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="button button-inactive Favorites-button"
>
            Favorites
            <span>
                         : 
            </span>
            0
</button>,
          "replace": false,
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites",
              <span>
                 : 
</span>,
              0,
            ],
            "className": "button button-inactive Favorites-button",
          },
          "ref": null,
          "rendered": Array [
            "Favorites",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": " : ",
              },
              "ref": null,
              "rendered": " : ",
              "type": "span",
            },
            0,
          ],
          "type": "button",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
